.github/workflows/website-snapshot.yml

```diff
@@ -0,0 +1,86 @@
+name: Build & Deploy spec to Github Pages
+
+on:
+  push:
+    branches:
+      - main
+      - docs-deploy # testing phase only
+
+permissions:
+  id-token: write
+
+jobs:
+  site-deploy:
+    name: Build and Publish To openlineage-site repo
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: 'source'
+      - uses: actions/checkout@v4
+        with:
+          repository: 'pawel-big-lebowski/openlineage-site' # TODO: convert to OpenLineage/openlineage-site
+          path: 'target'
+          token:  ${{ secrets.ACTIONS_GITHUB_TOKEN }}
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: "npm"
+          cache-dependency-path: source/website/package-lock.json
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'corretto'
+          java-version: '17'
+      - uses: shogo82148/actions-setup-perl@v1
+        with:
+          perl-version: "5.38"
+      - name: Regenerate Javadoc to website
+        run: |
+          cd source/client/java
+          ./gradlew javadoc
+          mkdir -p ../../website/static/apidocs/javadoc
+          cp -r build/docs/javadoc ../../website/static/apidocs
```

Comment:
I can see you've created script to regenerate docs. Why not use it here?

Reply:
The script was added when javadoc was generated with pre-commit which is no longer used. I will remove the script instead. Thanks for spotting this.